data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/model/operations/DescribeTargetEnvironmentRevisionRequest.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservicemodel.v1.model.operations;
+
+import lombok.Value;
+
+@Value
+public class DescribeTargetEnvironmentRevisionRequest {
+
+  private final String environmentArn;
```

Comment:
We should just take the environmentName and construct the id/arn internally

Reply:
Why? If ARNs are the canonical identifier for our resources, shouldn't we use them? In particular, we do need at least `(accountId, environmentName)` in order to make the correct determination here.

Reply:
Hm, thinking about this a bit more, we may want to put more/less in the EnvironmenID than is in the Arn (e.g. cluster ID). Do you think it's worth making an EnvironmentId value type? Then all the logic for converting to/from ARNs and such can live in there.

Reply:
Yes, this is already checked in the environmentId/arnGenerator